highfidelity/hifi
# WL#21166 Cloneable Equipment

https://worklist.net/21166

Adds four new grabbableKey fields that can be adjusted from the edit window

- cloneable: boolean value to allow for cloneable or non-cloneable.
- cloneLifetime: sets the life time for the clones
- cloneLimit: sets the max amount of clones you can have within a 50m radius
- cloneDynamic: Allows the cloned object to be dynamic

![new edit window](http://www.norteclabs.com/HF/test/Cloneable.jpg)

To test you will need hand controllers, Vive or Oculus Touch:

# Test Prep

1. Rez A Bow, Pingpong gun, Flaregun, or hookshotgun from the the marketplace
2. Set Cloneable on the objects
3. Dynamic and Grabbable should go off (edit.js only)
4. Turn off Edit window

# General Clonable Test 

1. Attempt to far grab object. It should not allow you to drag the source.
2. Go near the source and grab it with the trigger (or grips, but hold grips),
 a new clone of object should equip like any other object would.
3. You should be able to fire / use the object  
4. You should be able to drop object.

#  Limit Test

Suggest setting Cloneable dynamic, and gravity on source, then use the grip buttons to clone and throw the clone off as many times until it spawns none.

You  should only be able to clone the amount of clones that is set in cloneLimit.  Default, if not defined in the userdata, should be 10 units

# Lifetime Test

All cloned objects should have a lifetime and should expire eventually. This is set according to cloneLifetime. Default, if not defined in the userdata, should be 300s.

# Lock Test

You should have lock permissions: locking the entity should still allow others to interact and grab a clone.
